telemetry/src/metrics.rs

```diff
@@ -0,0 +1,174 @@
+//! [`Metrics`] and [`Status`]-related logic and functions.
+use std::{
+    sync::Arc,
+    time::{Duration, SystemTime},
+};
+
+use prometheus::{
+    core::{AtomicU64, GenericGauge, GenericGaugeVec},
+    Encoder, Histogram, HistogramOpts, HistogramVec, IntCounter, IntCounterVec, Opts, Registry,
+};
+use serde::{Deserialize, Serialize};
+
+/// Thin wrapper around duration that `impl`s [`Default`]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct Uptime(Duration);
+
+impl Default for Uptime {
+    fn default() -> Self {
+        Self(Duration::from_millis(0))
+    }
+}
+
+/// Response body for GET status request
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
+pub struct Status {
+    /// Number of connected peers, except for the reporting peer itself
+    pub peers: u64,
+    /// Number of committed blocks
+    pub blocks: u64,
+    /// Number of transactions committed in the last block
+    pub txs: u64,
+    /// Uptime since genesis block creation
+    pub uptime: Uptime,
+}
+
+impl From<&Arc<Metrics>> for Status {
+    fn from(val: &Arc<Metrics>) -> Self {
+        Self {
+            peers: val.connected_peers.get(),
+            blocks: val.block_height.get(),
+            txs: val.txs.with_label_values(&["total"]).get(),
+            uptime: Uptime(Duration::from_millis(val.uptime_since_genesis_ms.get())),
+        }
+    }
+}
+
+/// A strict superset of [`Status`].
```

Comment:
do we need separate `Status` anymore?

Reply:
Yeah. Serde.

Reply:
I don't get you, you can implement serde on metrics as well and only serialize those values from `Status`.
would it make sense to remove `Status` and keep only metrics? I mean, to open an issue to keep that in mind and simplify in the future

Reply:
#1773